### DISCUSSION TITLE
Conflicting TS guidance in Zustand's docs?

### DISCUSSION THREAD

**Discussion by rafegoldberg** [pmndrs/zustand#1743] [Q&A] [answered]
[Zustand's official TypeScript guide][docs] claims “that instead of writing `create(...)`, you have to write `create<T>()(...)` (notice the extra parenthesis `()` too along with the type parameter)”. [But this is *directly* contradicted by many TS examples throughout the rest of the docs](https://docs.pmnd.rs/zustand/guides/updating-state#flat-updates). 😬 The simpler examples seem to run fine in my experience, so I'm assuming it's the TypeScript guide that's out of date? Happy to PR an update to that doc, just wanted to double check first.

[docs]: https://docs.pmnd.rs/zustand/guides/typescript

**Comment by dai-shi** [comment, marked as answer]
Yeah, I think we should update docs to avoid confusion.
We need extra `()` only if we use some middleware with TypeScript, but it would be simpler to educate all TS users to use extra `()`.

Related: https://github.com/pmndrs/zustand/issues/1430#issuecomment-1373626898

**Reply by rafegoldberg** [reply]
Ahhh, gotchya, thanks for clarifying! I doubt I'm knowledgable enough to update the doc myself, so I'll leave any PR up to more experienced contributors. But I've marked this as answered; thanks again for clearing that up.
